src/main/java/org/sagebionetworks/web/client/presenter/TeamPresenter.java

```diff
@@ -124,43 +183,72 @@ public void onFailure(Throwable caught) {
 	@Override
 	public void leaveTeam() {
 		synAlert.clear();
-		String userId = authenticationController.getCurrentUserPrincipalId();
-		synapseClient.deleteTeamMember(userId, userId, team.getId(), new AsyncCallback<Void>() {
+		leaveTeamWidget.setRefreshCallback(new Callback() {

```

Comment:
Just create a single refreshCallback in the constructor, and set in leaveTeamWidget, inviteWidget, and editTeamWidget (once, in constructor)